src/bindings/python/PyImageDesc.h

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright Contributors to the OpenColorIO Project.
+
+#ifndef INCLUDED_OCIO_PYIMAGEDESC_H
+#define INCLUDED_OCIO_PYIMAGEDESC_H
+
+#include "PyOpenColorIO.h"
+#include "PyUtils.h"
+
+namespace OCIO_NAMESPACE
+{
+
+using PackedImageDescRcPtr = OCIO_SHARED_PTR<PackedImageDesc>;
+using PlanarImageDescRcPtr = OCIO_SHARED_PTR<PlanarImageDesc>;
+
+// ImageDesc does NOT claim ownership of its pixels or copy image data. This is problematic in 
+// Python when image data passed to the constructor only exists in the function call's scope. 
+// The py::buffer is garbage collected immediately following initialization, leaving a dangling 
+// pointer in the ImageDesc. Consider this scenario:
+//
+//   >>> img1 = OCIO.PackedImageDesc(np.array(0.18, ...), ...)
+//   >>> img1.getData()
+//   np.array(0.18, ...)
+//   >>> img2 = OCIO.PackedImageDesc(np.array(0.05, ...), ...)
+//   >>> img2.getData()
+//   np.array(0.05, ...)
+//   >>> img1.getData()
+//   np.array(0.05, ...)  # img1 and img2 are pointing to the same memory
+//
+// To get around this, ImageDesc and its subclasses are wrapped in a container struct which owns 
+// persistent copies of the initializing py::buffer object(s).
+
+struct OCIOHIDDEN PyImageDesc
```

Comment:
With C++11, the code can use `= default` for the constructor and the destructor, and add `noexcept` for all the other methods.

Reply:
Done.